src/actions/guides/database/deleting_records.cr

```diff
@@ -0,0 +1,169 @@
+class Guides::Database::DeletingRecords < GuideAction
+  guide_route "/database/deleting-records"
+
+  def self.title
+    "Deleting records"
+  end
+
+  def markdown : String
+    <<-MD
+    ## Deleting Records
+
+    ### Delete one
+
+    Deleting a single record is actually done on the [model](#{Guides::Database::Models.path}) directly.
+    Since each query returns an instance of the model, you can just call `delete` on that record.
+
+    ```crystal
+    user = UserQuery.find(4)
+
+    # DELETE FROM users WHERE users.id = 4
+    user.delete
+    ```
+
+    ### Bulk delete
+
+    If you need to bulk delete a group of records based on a where query, you can use `delete` at
+    the end of your query. This returns the number of records deleted.
+
+    ```crystal
+    # DELETE FROM users WHERE banned_at IS NOT NULL
+    UserQuery.new.banned_at.is_not_nil.delete
+    ```
+
+    ## Soft Deletes
+
+    A "soft delete" is when you want to hide a record as if it were deleted, but you want to keep the actual
+    record in your database. This allows you to restore the record without losing any previous data or associations.
+
+    Avram comes with some built-in modules to help working with soft deleted records a lot easier. Let's add it
+    to an existing `Article` model.
+
+    * We need to add a new `soft_deleted_at : Time?` column to our table that we want soft deletes.
+
+    ```
+    # Run this in your terminal
+    lucky gen.migration AddSoftDeleteToArticle
+    ```
+
+    * Open your new `db/migrations/#{Time.utc.to_s("%Y%m%d%H%I%S")}_add_soft_delete_to_article.cr` file.
+
+    ```crystal
+    def migrate
+      alter table_for(Article) do
+        add soft_deleted_at : Time?
+      end
+    end
+    ```
+
+    * Now open your `src/models/article.cr` file.
+
+    ```crystal
+    class Article < BaseModel
+      # Include this module
+      include Avram::SoftDelete::Model
+
+      table do
+        # Add this to your table
```

Comment:
```suggestion
        # Add the new column to your model
```

Since it is kind of already in the db table. So maybe this is clearer? Up to you